menu/menu.cpp

```diff
@@ -152,31 +134,93 @@ void menu::menu::subMenu() {
         case 4:
             std::cout << Term::move_cursor(17, 7)
                       << Term::color24_fg(255, 64, 0) << "    Press Q";
-            arrow = false;
+            settingsSub = true;
             anim::snakeSettings();
             break;
         case 5:
             std::cout << Term::move_cursor(17, 7)
                       << Term::color24_fg(255, 64, 0) << "    Press Q";
-            arrow = false;
+            selector = false;
             anim::snakeAbout();
             break;
         case 6:
             anim::snakeLicense();
-            arrow = false;
-            license_sub_open = true;
-            license_page_toggle = true;
-            anim::license_menu_toggle(license_page_toggle);
+            selector = false;
+            licenseSub = true;
+            licensePageToggle = true;
+            anim::license_menu_toggle(licensePageToggle);
             break;
     }
-    lastCursorState = cursorState;
+    if (!settingsSubSub)
+        lastCursorState = cursorState;
+    cursorState = 0;
+    updateDescription();
+    std::cout << Term::color(Term::fg::reset) << std::flush;
+}
+void menu::menu::handleGamemode() {
+    sub = false;
+    gamemodeSub = false;
+    draw::clearField();
+    snake::snake snakeObj;
+
+    switch (cursorState) {
+        case 0:  // SNEK
+            snakeObj.run();
+            break;
+        case 1:  // Classic
+            snakeObj.run();
+            break;
+        case 2:  // Infinite
+            snakeObj.run();
+            break;
+        case 3:  // Chaos
+            snakeObj.run();
+            break;
+        case 4:  // Creative
+            snakeObj.run();
+            break;
+    }
+
+    draw::snakeMenuBase();
     cursorState = 0;
     updateDescription();
+    std::cout << Term::move_cursor(9, 20) << Term::color24_fg(255, 64, 0)
+              << "<--" << Term::color(Term::fg::reset) << std::flush;
+}
+void menu::menu::handleSettings() {
+    settingsSub = false;
+    settingsSubSub = true;
+    draw::clearMenu();
+
+    switch (cursorState) {
+        case 0:
+            std::cout << Term::move_cursor(7, 10)
+                      << Term::color24_fg(255, 255, 0) << "CONTROLS!";
+            // updateDescription();
+            // anim::snakeControls();
```

Comment:
Yes. As soon as we add the control settings to this sub-menu, we have to update the settings description so that the player knows what the settings change in the game (like we did in other sub menus as well). And the `anim::snakeControls()`  function is for drawing this sub-menu's content (aka. settings).

In summary: Yes, it is TODO. I just commented it out, so that I exactly know where to continue.